src/bin/rustfmt.rs

```diff
@@ -9,17 +9,37 @@
 // except according to those terms.
 
 #![cfg(not(test))]
+#![feature(path_ext)]
 
 extern crate rustfmt;
 
 use rustfmt::{WriteMode, run};
 
-use std::fs::File;
-use std::io::Read;
+use std::env;
+use std::fs::{File, PathExt};
+use std::io::{self, Read, Error, ErrorKind};
+use std::path::PathBuf;
+
+fn lookup_config_file() -> io::Result<PathBuf> {
```

Comment:
Hey! Sorry for the delay.

I think that merging the files at the TOML level would be the easiest way of doing the merging since they're basically hashmaps. Then the merged TOML gets decoded into the config and we're done. I'm working on that now.